### ISSUE TITLE
Error when building conda environment

### ISSUE THREAD

**Issue by bblarsen-sci** [dms-vep/dms-vep-pipeline-3#163]
I upgraded the `dms-vep-pipeline-3` submodule to `v3.18` and am trying to build the new conda environment from `environment.yml.`

I am repeatedly getting this error during the build process.

```
Creating conda environment dms-vep-pipeline-3/environment.yml...
Downloading and installing remote packages.
CreateCondaEnvironmentException:
Could not create conda environment from /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/dms-vep-pipeline-3/environment.yml:
Command:
conda env create --quiet --no-default-packages --file "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_.yaml" --prefix "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_"
Output:
Channels:
 - conda-forge
 - bioconda
Platform: linux-64
Collecting package metadata (repodata.json): ...working... done
Solving environment: ...working... done
Preparing transaction: ...working... done
Verifying transaction: ...working... done
Executing transaction: ...working... done
Installing pip dependencies: ...working... Pip subprocess error:
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [171 lines of output]
      # pysam: cython is available - using cythonize if necessary
      # pysam: htslib mode is shared
      # pysam: HTSLIB_CONFIGURE_OPTIONS=None
      # pysam: (sysconfig) CC=gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat
      # pysam: (sysconfig) CFLAGS=-fno-strict-overflow -Wsign-compare -DNDEBUG -O2 -Wall -fPIC -O2   -isystem /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/include -fPIC -O2   -isystem /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/include  -fPIC
      # pysam: (sysconfig) LDFLAGS=-Wl,--allow-shlib-undefined -Wl,-rpath,/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib -Wl,-rpath-link,/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib -L/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib -Wl,--allow-shlib-undefined -Wl,-rpath,/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib -Wl,-rpath-link,/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib -L/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib
      checking for gcc... gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat accepts -g... yes
      checking for gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat option to accept ISO C89... none needed
      checking for ranlib... ranlib
      checking for grep that handles long lines and -e... /bin/grep
      checking for C compiler warning flags... unknown
      checking how to run the C preprocessor... gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat -E
      checking for egrep... /bin/grep -E
      checking for ANSI C header files... yes
      checking for sys/types.h... yes
      checking for sys/stat.h... yes
      checking for stdlib.h... yes
      checking for string.h... yes
      checking for memory.h... yes
      checking for strings.h... yes
      checking for inttypes.h... yes
      checking for stdint.h... yes
      checking for unistd.h... yes
      checking whether _XOPEN_SOURCE is declared... no
      checking C compiler flags needed for ssse3... -mssse3
      checking C compiler flags needed for popcnt... -mpopcnt
      checking C compiler flags needed for sse4.1... -msse4.1
      checking C compiler flags needed for avx2... -mavx2
      checking C compiler flags needed for avx512f... -mavx512f
      checking for pkg-config... /usr/bin/pkg-config
      checking pkg-config is at least version 0.9.0... yes
      checking for special C compiler options needed for large files... no
      checking for _FILE_OFFSET_BITS value needed for large files... no
      checking shared library type for unknown-Linux... plain .so
      checking whether the linker supports versioned symbols... no
      checking whether the compiler accepts -fvisibility=hidden... yes
      checking for stdlib.h... (cached) yes
      checking for unistd.h... (cached) yes
      checking for sys/param.h... yes
      checking for getpagesize... yes
      checking for working mmap... yes
      checking for gmtime_r... yes
      checking for fsync... yes
      checking for drand48... yes
      checking for srand48_deterministic... no
      checking whether fdatasync is declared... yes
      checking for fdatasync... yes
      checking for library containing log... -lm
      checking for zlib.h... yes
      checking for inflate in -lz... yes
      checking for library containing recv... none required
      checking for bzlib.h... yes
      checking for BZ2_bzBuffToBuffCompress in -lbz2... yes
      checking for lzma.h... no
      checking for lzma_easy_buffer_encode in -llzma... no
      configure: error: liblzma development files not found
      
      The CRAM format may use LZMA2 compression, which is implemented in HTSlib
      by using compression routines from liblzma <http://tukaani.org/xz/>.
      
      Building HTSlib requires liblzma development files to be installed on the
      build machine; you may need to ensure a package such as liblzma-dev (on Debian
      or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
      xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
      
      Either configure with --disable-lzma (which will make some CRAM files
      produced elsewhere unreadable) or resolve this error to build HTSlib.
      checking for gcc... gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat accepts -g... yes
      checking for gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat option to accept ISO C89... none needed
      checking for ranlib... ranlib
      checking for grep that handles long lines and -e... /bin/grep
      checking for C compiler warning flags... unknown
      checking how to run the C preprocessor... gcc -pthread -B /fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/compiler_compat -E
      checking for egrep... /bin/grep -E
      checking for ANSI C header files... yes
      checking for sys/types.h... yes
      checking for sys/stat.h... yes
      checking for stdlib.h... yes
      checking for string.h... yes
      checking for memory.h... yes
      checking for strings.h... yes
      checking for inttypes.h... yes
      checking for stdint.h... yes
      checking for unistd.h... yes
      checking whether _XOPEN_SOURCE is declared... no
      checking C compiler flags needed for ssse3... -mssse3
      checking C compiler flags needed for popcnt... -mpopcnt
      checking C compiler flags needed for sse4.1... -msse4.1
      checking C compiler flags needed for avx2... -mavx2
      checking C compiler flags needed for avx512f... -mavx512f
      checking for pkg-config... /usr/bin/pkg-config
      checking pkg-config is at least version 0.9.0... yes
      checking for special C compiler options needed for large files... no
      checking for _FILE_OFFSET_BITS value needed for large files... no
      checking shared library type for unknown-Linux... plain .so
      checking whether the linker supports versioned symbols... no
      checking whether the compiler accepts -fvisibility=hidden... yes
      checking for stdlib.h... (cached) yes
      checking for unistd.h... (cached) yes
      checking for sys/param.h... yes
      checking for getpagesize... yes
      checking for working mmap... yes
      checking for gmtime_r... yes
      checking for fsync... yes
      checking for drand48... yes
      checking for srand48_deterministic... no
      checking whether fdatasync is declared... yes
      checking for fdatasync... yes
      checking for library containing log... -lm
      checking for zlib.h... yes
      checking for inflate in -lz... yes
      checking for library containing recv... none required
      checking for bzlib.h... yes
      checking for BZ2_bzBuffToBuffCompress in -lbz2... yes
      checking for lzma.h... no
      checking for lzma_easy_buffer_encode in -llzma... no
      configure: error: liblzma development files not found
      
      The CRAM format may use LZMA2 compression, which is implemented in HTSlib
      by using compression routines from liblzma <http://tukaani.org/xz/>.
      
      Building HTSlib requires liblzma development files to be installed on the
      build machine; you may need to ensure a package such as liblzma-dev (on Debian
      or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
      xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
      
      Either configure with --disable-lzma (which will make some CRAM files
      produced elsewhere unreadable) or resolve this error to build HTSlib.
      Makefile:142: htscodecs.mk: No such file or directory
      config.mk:2: *** Resolve configure error first.  Stop.
      # pysam: htslib configure options: None
      Traceback (most recent call last):
        File "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/loc/scratch/2477890/pip-build-env-15164tqa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/loc/scratch/2477890/pip-build-env-15164tqa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/loc/scratch/2477890/pip-build-env-15164tqa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 522, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/loc/scratch/2477890/pip-build-env-15164tqa/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 442, in <module>
        File "<string>", line 82, in run_make_print_config
        File "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib/python3.12/subprocess.py", line 466, in check_output
          return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/fh/fast/bloom_j/computational_notebooks/blarsen/2024/Nipah_Malaysia_RBP_DMS_Sinai_mAbs/.snakemake/conda/9ddc148d62fc5895f9b9f9af40d8ae3d_/lib/python3.12/subprocess.py", line 571, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['make', '-s', 'print-config']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

failed

CondaEnvException: Pip failed
```

**Comment by jbloom** [comment]
I will look into this. By the way, I would probably **not** suggest doing this upgrade for the RBP paper though unless you really think it is important for some reason (???). The updated packages will very slightly change results in a way that might force you to remake all figures in a way that isn't worth it.

**Comment by bblarsen-sci** [comment]
It's not for the RBP paper, I am trying to upgrade the pipeline for my repo that is specifically for analyzing the sera escape data.

**Comment by jbloom** [comment]
Sounds good. I am making some other changes right now to address other issues, so will look into this as a finish those.

**Comment by jbloom** [comment]
@bblarsen-sci, for the environment specified in the `environment.yml` file for version 3.19 (which I will tag and post to GitHub soon), I am able to build the environment with no problem.

So can you test if things work for you with that version.

If not, can you post more details about the error including what version of `conda` you are using and the command you are using to build the environment.

**Comment by bblarsen-sci** [comment]
I just updated `dms-vep-pipeline-3` to newest version (3.19) and tried building the conda environment again. I got a similar error. I am using `conda` v24.7.1. Should I try upgrading conda?

```
Pip subprocess error:
  error: subprocess-exited-with-error
  
  × Getting requirements to build wheel did not run successfully.
  │ exit code: 1
  ╰─> [171 lines of output]
      # pysam: cython is available - using cythonize if necessary
      # pysam: htslib mode is shared
      # pysam: HTSLIB_CONFIGURE_OPTIONS=None
      # pysam: (sysconfig) CC=gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat
      # pysam: (sysconfig) CFLAGS=-fno-strict-overflow -Wsign-compare -DNDEBUG -O2 -Wall -fPIC -O2   -isystem /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/include -fPIC -O2   -isystem /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/include  -fPIC
      # pysam: (sysconfig) LDFLAGS=-Wl,--allow-shlib-undefined -Wl,-rpath,/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib -Wl,-rpath-link,/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib -L/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib -Wl,--allow-shlib-undefined -Wl,-rpath,/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib -Wl,-rpath-link,/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib -L/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib
      checking for gcc... gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat accepts -g... yes
      checking for gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat option to accept ISO C89... none needed
      checking for ranlib... ranlib
      checking for grep that handles long lines and -e... /bin/grep
      checking for C compiler warning flags... unknown
      checking how to run the C preprocessor... gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat -E
      checking for egrep... /bin/grep -E
      checking for ANSI C header files... yes
      checking for sys/types.h... yes
      checking for sys/stat.h... yes
      checking for stdlib.h... yes
      checking for string.h... yes
      checking for memory.h... yes
      checking for strings.h... yes
      checking for inttypes.h... yes
      checking for stdint.h... yes
      checking for unistd.h... yes
      checking whether _XOPEN_SOURCE is declared... no
      checking C compiler flags needed for ssse3... -mssse3
      checking C compiler flags needed for popcnt... -mpopcnt
      checking C compiler flags needed for sse4.1... -msse4.1
      checking C compiler flags needed for avx2... -mavx2
      checking C compiler flags needed for avx512f... -mavx512f
      checking for pkg-config... /usr/bin/pkg-config
      checking pkg-config is at least version 0.9.0... yes
      checking for special C compiler options needed for large files... no
      checking for _FILE_OFFSET_BITS value needed for large files... no
      checking shared library type for unknown-Linux... plain .so
      checking whether the linker supports versioned symbols... no
      checking whether the compiler accepts -fvisibility=hidden... yes
      checking for stdlib.h... (cached) yes
      checking for unistd.h... (cached) yes
      checking for sys/param.h... yes
      checking for getpagesize... yes
      checking for working mmap... yes
      checking for gmtime_r... yes
      checking for fsync... yes
      checking for drand48... yes
      checking for srand48_deterministic... no
      checking whether fdatasync is declared... yes
      checking for fdatasync... yes
      checking for library containing log... -lm
      checking for zlib.h... yes
      checking for inflate in -lz... yes
      checking for library containing recv... none required
      checking for bzlib.h... yes
      checking for BZ2_bzBuffToBuffCompress in -lbz2... yes
      checking for lzma.h... no
      checking for lzma_easy_buffer_encode in -llzma... no
      configure: error: liblzma development files not found
      
      The CRAM format may use LZMA2 compression, which is implemented in HTSlib
      by using compression routines from liblzma <http://tukaani.org/xz/>.
      
      Building HTSlib requires liblzma development files to be installed on the
      build machine; you may need to ensure a package such as liblzma-dev (on Debian
      or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
      xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
      
      Either configure with --disable-lzma (which will make some CRAM files
      produced elsewhere unreadable) or resolve this error to build HTSlib.
      checking for gcc... gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat
      checking whether the C compiler works... yes
      checking for C compiler default output file name... a.out
      checking for suffix of executables...
      checking whether we are cross compiling... no
      checking for suffix of object files... o
      checking whether we are using the GNU C compiler... yes
      checking whether gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat accepts -g... yes
      checking for gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat option to accept ISO C89... none needed
      checking for ranlib... ranlib
      checking for grep that handles long lines and -e... /bin/grep
      checking for C compiler warning flags... unknown
      checking how to run the C preprocessor... gcc -pthread -B /home/blarsen/miniforge3/envs/dms-vep-pipeline-3/compiler_compat -E
      checking for egrep... /bin/grep -E
      checking for ANSI C header files... yes
      checking for sys/types.h... yes
      checking for sys/stat.h... yes
      checking for stdlib.h... yes
      checking for string.h... yes
      checking for memory.h... yes
      checking for strings.h... yes
      checking for inttypes.h... yes
      checking for stdint.h... yes
      checking for unistd.h... yes
      checking whether _XOPEN_SOURCE is declared... no
      checking C compiler flags needed for ssse3... -mssse3
      checking C compiler flags needed for popcnt... -mpopcnt
      checking C compiler flags needed for sse4.1... -msse4.1
      checking C compiler flags needed for avx2... -mavx2
      checking C compiler flags needed for avx512f... -mavx512f
      checking for pkg-config... /usr/bin/pkg-config
      checking pkg-config is at least version 0.9.0... yes
      checking for special C compiler options needed for large files... no
      checking for _FILE_OFFSET_BITS value needed for large files... no
      checking shared library type for unknown-Linux... plain .so
      checking whether the linker supports versioned symbols... no
      checking whether the compiler accepts -fvisibility=hidden... yes
      checking for stdlib.h... (cached) yes
      checking for unistd.h... (cached) yes
      checking for sys/param.h... yes
      checking for getpagesize... yes
      checking for working mmap... yes
      checking for gmtime_r... yes
      checking for fsync... yes
      checking for drand48... yes
      checking for srand48_deterministic... no
      checking whether fdatasync is declared... yes
      checking for fdatasync... yes
      checking for library containing log... -lm
      checking for zlib.h... yes
      checking for inflate in -lz... yes
      checking for library containing recv... none required
      checking for bzlib.h... yes
      checking for BZ2_bzBuffToBuffCompress in -lbz2... yes
      checking for lzma.h... no
      checking for lzma_easy_buffer_encode in -llzma... no
      configure: error: liblzma development files not found
      
      The CRAM format may use LZMA2 compression, which is implemented in HTSlib
      by using compression routines from liblzma <http://tukaani.org/xz/>.
      
      Building HTSlib requires liblzma development files to be installed on the
      build machine; you may need to ensure a package such as liblzma-dev (on Debian
      or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
      xz (via Homebrew on macOS) is installed; or build XZ Utils from source.
      
      Either configure with --disable-lzma (which will make some CRAM files
      produced elsewhere unreadable) or resolve this error to build HTSlib.
      Makefile:142: htscodecs.mk: No such file or directory
      config.mk:2: *** Resolve configure error first.  Stop.
      # pysam: htslib configure options: None
      Traceback (most recent call last):
        File "/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 353, in <module>
          main()
        File "/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 335, in main
          json_out['return_val'] = hook(**hook_input['kwargs'])
                                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib/python3.12/site-packages/pip/_vendor/pyproject_hooks/_in_process/_in_process.py", line 118, in get_requires_for_build_wheel
          return hook(config_settings)
                 ^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-4tun504r/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 334, in get_requires_for_build_wheel
          return self._get_build_requires(config_settings, requirements=[])
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/tmp/pip-build-env-4tun504r/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 304, in _get_build_requires
          self.run_setup()
        File "/tmp/pip-build-env-4tun504r/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 522, in run_setup
          super().run_setup(setup_script=setup_script)
        File "/tmp/pip-build-env-4tun504r/overlay/lib/python3.12/site-packages/setuptools/build_meta.py", line 320, in run_setup
          exec(code, locals())
        File "<string>", line 442, in <module>
        File "<string>", line 82, in run_make_print_config
        File "/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib/python3.12/subprocess.py", line 466, in check_output
          return run(*popenargs, stdout=PIPE, timeout=timeout, check=True,
                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
        File "/home/blarsen/miniforge3/envs/dms-vep-pipeline-3/lib/python3.12/subprocess.py", line 571, in run
          raise CalledProcessError(retcode, process.args,
      subprocess.CalledProcessError: Command '['make', '-s', 'print-config']' returned non-zero exit status 2.
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× Getting requirements to build wheel did not run successfully.
│ exit code: 1
╰─> See above for output.

note: This error originates from a subprocess, and is likely not a problem with pip.

failed

CondaEnvException: Pip failed


```

**Comment by Bernadetadad** [comment]
I'm getting the same error using `conda 24.9.2`

**Comment by bblarsen-sci** [comment]
I tried running `conda clean --all`, that still didn't help. 

Next, I tried changing python version from 3.12 to 3.11 in the `environment.yml` file, and now it is building correctly. I'm not exactly sure why?

**Comment by bblarsen-sci** [comment]
As far as I can tell, it appears the issue is with `pysam`, which is a dependency of `alignparse`. Alignparse is [requiring](https://github.com/jbloomlab/alignparse/blob/master/setup.py) an older version of [pysam](https://pysam.readthedocs.io/en/latest/release.html) for some reason. I'm not sure if that would fix it, or why @jbloom can build the environment, but @Bernadetadad and I cannot.

**Comment by WillHannon-MCB** [comment]
@jbloom, are you making the environment on `rhino`?

**Comment by jbloom** [comment]
Yes, I built it on rhino

**Comment by WillHannon-MCB** [comment]
Adding [`xz`](https://anaconda.org/conda-forge/xz) to `environment.yml` fixes the issue. Here's my theory:

As @bblarsen-sci pointed out, it's an issue with [`pysam`](https://pysam.readthedocs.io/en/latest/release.html), a dependency of `alignparse`. When `pip` attempts to build the wheel for `pysam`, it checks for the development version of a package called `liblzma`. It's important for decompressing sequencing file formats like CRAM.

```txt
checking for lzma.h... no <------------------------------------- pip looks for this header file
checking for lzma_easy_buffer_encode in -llzma... no
configure: error: liblzma development files not found

The CRAM format may use LZMA2 compression, which is implemented in HTSlib
by using compression routines from liblzma <http://tukaani.org/xz/>.

Building HTSlib requires liblzma development files to be installed on the
build machine; you may need to ensure a package such as liblzma-dev (on Debian
or Ubuntu Linux), xz-devel (on RPM-based Linux distributions or Cygwin), or
xz (via Homebrew on macOS) is installed; or build XZ Utils from source. 
```

@jbloom would have no issue building the environment if he installed the `liblzma` development package locally at some point.

@bblarsen-sci could build the environment by lowering the Python version from 3.12 to 3.11 if a pre-built `pysam` wheel including `lzma.h` exists for Python 3.11 on PyPI. I haven't confirmed that is the case, but `pip` is certainly building the wheel for Python 3.12:

```txt
Building wheel for pysam (pyproject.toml): started
Building wheel for pysam (pyproject.toml): still running...
Building wheel for pysam (pyproject.toml): still running...
Building wheel for pysam (pyproject.toml): still running...
Building wheel for pysam (pyproject.toml): finished with status 'done'
```

By adding [`xz`](https://anaconda.org/conda-forge/xz) to `environment.yml`, we're making sure that the development version of `liblzma` is installed if the pre-bulit `pysam` wheel isn't available from PyPI. Essentially, what the warning from `pysam` suggests.

**Comment by jbloom** [comment]
@bblarsen-sci @Bernadetadad, the solution suggested by @WillHannon-MCB is now implemented in version 3.19.1. If that still does not fix the problem, re-open this issue w details.
